### DISCUSSION TITLE
CsvHelper working differently with TextDelimiter & FieldDelimiter

### DISCUSSION THREAD

**Discussion by gmdevan** [JoshClose/CsvHelper#2149] [Q&A] [answered]
Hi Team,
We are using CsvHelper for processing csv & dat files. For the dat files processing, we are keeping the "TextDelimiter": "þ" & "FieldDelimiter": "¶"

If the dat file is NOT having any double quotes ('"'), then the csvhelper is working fine. If the dat file is having a double quotes ('"'), then the csvhelper is splitting the data incorrectly.

```
using (var reader = new StreamReader(datFilePath))
using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
{
    if (LoadFileExtn.ToLower() == ".dat")
    {
        //csv.Configuration.IgnoreQuotes = true;
        csv.Configuration.Delimiter = "¶";
        csv.Configuration.Quote = "þ";
    }
    else
    {
        csv.Configuration.Delimiter = "|";
    }
    csv.Read();
    csv.ReadHeader();
    List<IDictionary<string, object>> dataRecords = csv.GetRecords<dynamic>()
                   .Select(x => (IDictionary<string, object>)x)
                   .ToList();
    foreach (var record in dataRecords)
    {
    
    }
}
```
**First Row:**
![image](https://user-images.githubusercontent.com/68700481/232416987-58a97d11-1d97-4d1e-9aac-156b1707a017.png)

**Second Row**
![image](https://user-images.githubusercontent.com/68700481/232417096-63a62bfa-b6af-4965-b0f7-649f2a656a63.png)

Sample load file link --> https://1drv.ms/u/s!ArE6Z5QM9W7HpFWc_TYlIPK1mPA7?e=ViD6zd

**Comment by AltruCoder** [comment, marked as answer]
Question was answered on StackOverflow https://stackoverflow.com/a/76035266/2355006

Need to also set the `Escape` character.

`csv.Configuration.Escape = 'þ';`
